_resources/port1.0/group/gobject_introspection-1.0.tcl

```diff
@@ -73,6 +73,16 @@ pre-build {
             build.args-append       CC="${configure.cc} [join ${configure.cc_archflags}]"
         }
 
+        # see https://trac.macports.org/ticket/59078
+        # -isysroot cannot be empty (see _osx_support.py in python installation)
+        if {${configure.sdkroot} != ""} {
+            set sdk_root "${configure.sdkroot}"
+        } else {
+            set sdk_root "/"
```

Comment:
I am wondering if this needs to be a bit more fancy than just "/". I think many systems now, and most in the future, will not have a functioning sysroot at "/". Should we run `xcrun --show-sdk-path` here, and then fall back to "/" if that is empty?

And, TBH,  I think this should all be in base (I have a PR working on that idea now).

Reply:
Isn't most of that logic (including running `xcrun`) taken care of [in the base code](https://github.com/macports/macports-base/blob/427ce2062d66581788a75aba625681cc79f3fbe3/src/port1.0/portconfigure.tcl#L457) when setting `configure.sdkroot`?

The solution here is similar to the [code](https://github.com/macports/macports-ports/blob/07fe724ae90c80251c4d07fe0aeb1eab19baffb8/_resources/port1.0/group/cmake-1.1.tcl#L439) in the cmake 1.1 PG.

Reply:
Right .. I see. Assuming configure is actually run, by that code ${configure.sdkroot} should never be ""... but "/" is a fallback just in case configure is not executed, or similar unknown circumstances I guess. 

We should indeed make sure, somehow in base, that ${configure.sdkroot} can never be "", whether configure is run or not, and then condense all our logic into using that (and hopefully never backing the sdkroot into any build files any more).

This macOS ${configure.sdkroot} business is bubbling up to be one of our major headaches at present, and clearly we are not alone in that (cmake and gcc both have active issues/bug reports going on about it). Thanks for taking the time.